code/game/mecha/equipment/tools/medical_tools.dm

```diff
@@ -404,245 +404,293 @@
+
+/obj/item/mecha_parts/mecha_equipment/tool/syringe_gun/proc/get_reagents_list()
+	var/output
+	for(var/i=1 to known_reagents.len)
+		var/reagent_id = known_reagents[i]
+		output += {"<input type="checkbox" value="[reagent_id]" name="reagent_[i]" [(reagent_id in processed_reagents)? "checked=\"1\"" : null]> [known_reagents[reagent_id]]<br />"}
+	return output
+
+/obj/item/mecha_parts/mecha_equipment/tool/syringe_gun/proc/get_current_reagents()
+	var/output
+	for(var/datum/reagent/R in reagents.reagent_list)
+		if(R.volume > 0)
+			output += "[R]: [round(R.volume,0.001)] - <a href=\"?src=\ref[src];purge_reagent=[R.id]\">Purge Reagent</a><br />"
+	if(output)
+		output += "Total: [round(reagents.total_volume,0.001)]/[reagents.maximum_volume] - <a href=\"?src=\ref[src];purge_all=1\">Purge All</a>"
+	return output || "None"
+
+/obj/item/mecha_parts/mecha_equipment/tool/syringe_gun/proc/load_syringe(obj/item/weapon/reagent_containers/syringe/S)
+	if(syringes.len<max_syringes)
+		if(get_dist(src,S) >= 2)
+			occupant_message("The syringe is too far away.")
 			return 0
-		occupant_message("Analyzing reagents...")
-		for(var/datum/reagent/R in A.reagents.reagent_list)
-			if(R.reagent_state == 2 && add_known_reagent(R.id,R.name))
-				occupant_message("Reagent analyzed, identified as [R.name] and added to database.")
-				send_byjax(chassis.occupant,"msyringegun.browser","reagents_form",get_reagents_form())
-		occupant_message("Analyzis complete.")
+		for(var/obj/structure/D in S.loc)//Basic level check for structures in the way (Like grilles and windows)
+			if(!(D.CanPass(S,src.loc)))
+				occupant_message("Unable to load syringe.")
+				return 0
+		for(var/obj/machinery/door/D in S.loc)//Checks for doors
+			if(!(D.CanPass(S,src.loc)))
+				occupant_message("Unable to load syringe.")
+				return 0
+		S.reagents.trans_to_obj(src, S.reagents.total_volume)
+		S.forceMove(src)
+		syringes += S
+		occupant_message("Syringe loaded.")
+		update_equip_info()
 		return 1
+	occupant_message("The [src] syringe chamber is full.")
+	return 0
 
-	proc/add_known_reagent(r_id,r_name)
-		set_ready_state(0)
-		do_after_cooldown()
-		if(!(r_id in known_reagents))
-			known_reagents += r_id
-			known_reagents[r_id] = r_name
-			return 1
+/obj/item/mecha_parts/mecha_equipment/tool/syringe_gun/proc/analyze_reagents(atom/A)
+	if(get_dist(src,A) >= 4)
+		occupant_message("The object is too far away.")
+		return 0
+	if(!A.reagents || istype(A,/mob))
+		occupant_message("<span class=\"alert\">No reagent info gained from [A].</span>")
 		return 0
+	occupant_message("Analyzing reagents...")
+	for(var/datum/reagent/R in A.reagents.reagent_list)
+		if(R.reagent_state == 2 && add_known_reagent(R.id,R.name))
+			occupant_message("Reagent analyzed, identified as [R.name] and added to database.")
+			send_byjax(chassis.occupant,"msyringegun.browser","reagents_form",get_reagents_form())
+		else
+			occupant_message("Reagent \"[R.name]\" too complex to analyze, skipping.")

```

Comment:
I believe this message will print if you try to analyze a container of something you already analyzed.   So click twice on something and it will say
_Reagent analyzed, identified as milk and added to database._
_Reagent "milk" too complex to analyze, skipping_

Reply:
Ok, altered the branching there. How's that?